### DISCUSSION TITLE
Welcome to Grafana-Google-Analytics-DataSource Discussions!

### DISCUSSION THREAD

**Discussion by lcc3108** [blackcowmoo/Grafana-Google-Analytics-DataSource#30] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by G1K** [comment]
Hey! @lcc3108  Thanks for your plugin, it works!
I'm wondering if there is an opportunity to add:

1) Selecting the or time shift period in the query. To implement comparison in periods in one chart.

2) Castom field to add filtering to request.

3) Backdrops for supporting the alerts for the data source.

**Reply by lcc3108** [reply]
Thank you for using our Grafana plugin

It would be nice to know more about your opinion.

1. Do you want to use multiple time range each panel?
Or do you want to use multiple time range each query on panel

2. Custom Field mean Custom Dimensions and Custom Metrics of Custom Definition in the GA menu?
   If you know how to import through GA API or have a link, please let me know.

3. Periodic data fetch is required for alert.
    plugin is fetching data through the GA API
    GA API has default resource quota (you can increase it to request at gcp).
    - Day : 50,000
    - Minute: 1,200

   If the resource quota is exceeded, there is a problem importing the data.

**Reply by G1K** [reply]
1) I would like to be able to redefine the time interval or make it relative in the context of the query

2) No, I don't mean custom metrics, custom dimensions right now. I would like to be able to add filtering to the query. For example medium = cpc, country not in ( ar, ab, an )
https://developers.google.com/analytics/devguides/reporting/core/v4/basics?hl=en#filtering

3) The data in analytics reports is not updated very often. Therefore, for most schedules, a request once per minute, and sometimes 5-10 minutes, will be acceptable. One such request per day will consume less than 1440 limits, but will allow, for example, to make notifications in case of large deviations relative to the last week or month.

**Reply by lcc3108** [reply]
Thank you for your answer.

I will work number 2 first

I'll think about other improvements a little more.

**Comment by fsantosvwfs** [comment]
Hello guys! How are you?

Thank you for the plugin.

I have a question about the connectivity: Does the plugin support a http Proxy to connect to the Google Analytics APIs? 

Thanks in advance.

Best Regards,
Felipe

**Reply by lcc3108** [reply]
The http proxy is not supported.
Do you have a use case?

**Reply by fsantosvwfs** [reply]
Hello!

Yes, I don't have local internet access.
All internet access must be through a http proxy so the plugin does not work in our case.

Thanks for the answer! :)

Best regards.

**Reply by lcc3108** [reply]
Thank you for your opinion.
However, what you requested is a special case and only a few people will experience it.

I will create the issue, but I think the priority will be low.
Of course, PR after modification is highly welcome.

Thank you.

**Reply by fsantosvwfs** [reply]
Nice, thank you! :)
